src/Temporalio/Client/ITemporalClientPlugin.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Interface for temporal client plugins.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: This API is experimental and may change in the future.
+    /// </remarks>
+    public interface ITemporalClientPlugin
+    {
+        /// <summary>
+        /// Gets the plugin name.
+        /// </summary>
+        public string Name { get; }
```

Comment:
```suggestion
        string Name { get; }
```

`public` is implied for all interface things